dpath/__init__.py

```diff
@@ -45,7 +49,22 @@ def _split_path(path: Path, separator: Optional[str] = "/") -> Union[List[PathSe
     else:
         split_segments = path.lstrip(separator).split(separator)
 
-    return split_segments
+    final = []
```

Comment:
Please use more meaningful variable names and/or inline documentation as needed

Reply:
The remarks here made me revise this function:
1. added comments and expanded docstring
2. 'final' now becomes  'split_compiled_segments'
3. *type annotations gave me some trouble*; since I had decided **not to change 'Path' to introduce Pattern**. I felt that doing it would have too wide impact.
  - added type GlobExtend which extends Glob by adding Pattern
  - made _split_path `_split_path(path: GlobExtend,...)` to allow for Patterns and Globs.
  - added test to avoid trying to apply lstrip to ints or bools, main benefit raising ValueError with a meaningful message. In some sense this checks that requirement in the type annotation for arg. path is met.
  - allowed Pattern as a leaf, this has the advantage of treating similarly `dpath.search(x, "{regex}" ...)`and `dpath.search(x, re.compile("regex"),...)` (and `dpath.search(x, [ re.compile("regex")],...)` which is the list form with length 1)
  
This way, the type annotation corresponds to what the function expects, and I hope everything is clearer.